tripal_chado/src/Form/ChadoNewPubSearchQueryForm.php

```diff
@@ -0,0 +1,676 @@
+      )->toString() . '</li>';
+    $html .= '</ul>';
+    $form['new_publication_link'] = [
+      '#markup' => $html
+    ];
+    unset($html);
+
+    // Show the list of importers available for user to select
+    $form = $this->form_elements_importer_selection($form, $form_state);
+
+    // If the button_next was clicked, it will exist in the form_state_values
+    if (isset($form_state_values['button_next']) || $pub_import_id != null) {
+      // Once clicked, hide the 'next' button by changing type to hidden
+      $form['button_next']['#type'] = 'hidden';
+
+      // Disable the click radio options
+      $form['plugin_id']['#attributes'] = array('onclick' => 'return false;');
+
+
+      // add the elements for the specific importer (below function initialized plugin and calls form function)
+      $form = $this->form_elements_specific_importer($form, $form_state);
+
+      // add the common elements (like search criteria)
+      $form = $this->form_elements_common($form, $form_state);
+      
+      // handle previous user input
+      if ($pub_import_id != 'null') {
+        $this->form_elements_load_previous_user_input(
+          $this->form_state_previous_user_input, $form['pub_library']
+        );
+      }
+
+      // If the test button was clicked - run the TripalPubLibrary Plugin specific test function
+      if (isset($_SESSION['tripal_pub_import'])) {
+        if ($_SESSION['tripal_pub_import']['perform_test'] == 1) {
+          $plugin_id = $form['plugin_id']['#default_value'];
+          if ($plugin_id) {
+            // Instantiate the selected plugin
+            // Pub Library Manager is found in tripal module: 
+            // tripal/tripal/src/TripalPubLibrary/PluginManagers/TripalPubLibraryManager.php
+            $pub_library_manager = \Drupal::service('tripal.pub_library');
+            $plugin = $pub_library_manager->createInstance($plugin_id, []);
+
+            // The selected plugin defines a test specific to itself.
+            $criteria_column_array = $_SESSION['tripal_pub_import']['perform_test_criteria_array'];
+            $results = $plugin->test($form, $form_state, $criteria_column_array);
+
+            // On successful results, it should return array with keys total_records, search_str, pubs(array)
+            $headers = ['', 'Publication', 'Authors'];
+            $form['test_results_table'] = [
+              '#type' => 'table',
+              '#header' => $headers,
+              '#prefix' => '<div id="test_results_table">',
+              '#suffix' => '</div>',
+              '#weight' => 1000, // arbitrary heavier number so table is below most options
+            ];
+
+            if ($results != NULL) {
+
+              $form['test_results_count_info'] = [
+                '#markup' => '<h1>Test results</h1><div>Found ' . $results['total_records'] . 
+                  ' publications. Showing the first 5 publications.</div>',
```

Comment:
```suggestion
                  ' publications.' . ($results['total_records']>5?' Showing the first 5 publications.':'') . '</div>',
```
Just a check so if there are 5 or fewer search hits, we don't display `Showing the first 5 publications`